### PR TITLE
Upload Code Dailog with a code editor

### LINKED ISSUE #8: FE: Upload code dialog
- a material-ui dialog (https://material-ui.com/components/dialogs/) to upload code and start a review 
- create a rich text editor that can be used across the application that has some syntax highlighting for different languages (consider draftjs + prism, or a full on code editor, etc.)

### PR DESCRIPTION
**Pull request by abhishekchohan**
**What it does ?**
- Implements the Dialog box reusable component.
- User can select the language from dropdown.
- Dialog box contains 1 code editor to edit code (That highlights the selected language syntax)
- Comment box that take user input for comments if any for the added code.
- this PR implements new custom scrollbars throughout the app.
- Some code refactoring..etc..

**What to avoid?**
- Nothing except you will see 2 text editor but one is just a code snippet displayed with syntax highlighting.
- This is done to have a boilerplate code ready for future implementations in other pages.

**Lastly: Navbar can use this reusable component instead of the hard coded button now.**

**Some screenshots**

component_button

select langauge

code editor with highlighter

no-editable-codeSnippet

### CODE REVIEW

File: client/src/components/UploadCodeDailog.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+import Button from "@material-ui/core/Button";
+import { Dialog, makeStyles, TextField, Typography } from "@material-ui/core";
+import CodeMirror from "@uiw/react-codemirror";
+import "codemirror/keymap/sublime";
+import "codemirror/theme/monokai.css";
+
+//modes ==> tells the codemirror library about the kind of
+//file we are opening so that respective syntax highlighting can be achieved.
+
+const modes = [
```
Comment (skassam21) [inline client/src/components/UploadCodeDailog.js:11]: can we extract this into a constants file - as we will probably be using this in multiple places

File: client/src/components/UploadCodeDailog.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  { label: "JavaScript", value: "jsx" },
+  { label: "CSS", value: "css" },
+  { label: "JAVA", value: "java" },
```
Comment (skassam21) [inline client/src/components/UploadCodeDailog.js:14]: Should be "Java"

**Review by skassam21 — COMMENTED** [review]
Can you make it look closer to the spec? Here are a couple of comments:

1. No need to have the comment div under the Submit button
2. You will need a Title TextField object 
3. Can you add some margin between the comments and the codeMirror, and the selection dropdown?

Did you want to try to add RichText to the textField?

File: client/src/components/UploadCodeDailog.js (hunk 2 above)
Comment (abhishekchohan) [reply client/src/components/UploadCodeDailog.js:14]: Fixed.

File: client/src/components/UploadCodeDailog.js (hunk 1 above)
Comment (abhishekchohan) [reply client/src/components/UploadCodeDailog.js:11]: Done as suggested

**Comment by abhishekchohan** [comment]
> Can you make it look closer to the spec? Here are a couple of comments:
> 
> 1. No need to have the comment div under the Submit button
> 2. You will need a Title TextField object
> 3. Can you add some margin between the comments and the codeMirror, and the selection dropdown?
> 
> Did you want to try to add RichText to the textField?


1. commentDiv was just a placeholder but I removed it.
2. ok sure. Working on it.
3. I added the margins as suggested.

yes I tried that yesterday but found this module little easy so implemented that but no worries, now I will start working on the rich text instead.

Thanks for the feedback

**Pull request closed without merging**